gobblin-modules/gobblin-orc/src/main/java/org/apache/gobblin/writer/GobblinOrcWriter.java

```diff
@@ -0,0 +1,264 @@
+import org.apache.hadoop.hive.ql.exec.vector.BytesColumnVector;
+import org.apache.hadoop.hive.ql.exec.vector.ColumnVector;
+import org.apache.hadoop.hive.ql.exec.vector.DecimalColumnVector;
+import org.apache.hadoop.hive.ql.exec.vector.DoubleColumnVector;
+import org.apache.hadoop.hive.ql.exec.vector.ListColumnVector;
+import org.apache.hadoop.hive.ql.exec.vector.LongColumnVector;
+import org.apache.hadoop.hive.ql.exec.vector.MapColumnVector;
+import org.apache.hadoop.hive.ql.exec.vector.StructColumnVector;
+import org.apache.hadoop.hive.ql.exec.vector.UnionColumnVector;
+import org.apache.hadoop.hive.ql.exec.vector.VectorizedRowBatch;
+import org.apache.hadoop.hive.serde2.SerDeException;
+import org.apache.hadoop.hive.serde2.avro.AvroObjectInspectorGenerator;
+import org.apache.hadoop.hive.serde2.avro.AvroSerdeUtils;
+import org.apache.orc.OrcFile;
+import org.apache.orc.TypeDescription;
+import org.apache.orc.Writer;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.gobblin.configuration.State;
+import org.apache.gobblin.state.ConstructState;
+
+
+/**
+ * A wrapper for ORC-core writer without dependency on Hive SerDe library.
+ */
+@Slf4j
+public class GobblinOrcWriter extends FsDataWriter<GenericRecord> {
+  static final String ORC_WRITER_PREFIX = "orcWriter.";
+  private static final String ORC_WRITER_BATCH_SIZE = ORC_WRITER_PREFIX + "batchSize";
+  private static final int DEFAULT_ORC_WRITER_BATCH_SIZE = 1000;
+
+  /**
+   * Check comment of {@link #deepCleanRowBatch} for the usage of this configuration.
+   */
+  private static final String ORC_WRITER_DEEP_CLEAN_EVERY_BATCH = ORC_WRITER_PREFIX + "deepCleanBatch";
+
+  private final GenericRecordToOrcValueWriter valueWriter;
+  @VisibleForTesting
+  final VectorizedRowBatch rowBatch;
+  private final Writer orcFileWriter;
+
+  // the close method may be invoked multiple times, but the underlying writer only supports close being called once
+  private volatile boolean closed = false;
+  private final boolean deepCleanBatch;
+
+  private final int batchSize;
+  private final Schema avroSchema;
+
+  public GobblinOrcWriter(FsDataWriterBuilder<Schema, GenericRecord> builder, State properties)
+      throws IOException {
+    super(builder, properties);
+
+    log.info("Start to construct a ORC-Native Writer");
+
+    // Create value-writer which is essentially a record-by-record-converter with buffering in batch.
+    this.avroSchema = builder.getSchema();
+    TypeDescription typeDescription = AvroOrcSchemaConverter.getOrcSchema(this.avroSchema);
+    this.valueWriter = new GenericRecordToOrcValueWriter(typeDescription, this.avroSchema, properties);
+    this.batchSize = properties.getPropAsInt(ORC_WRITER_BATCH_SIZE, DEFAULT_ORC_WRITER_BATCH_SIZE);
+    this.rowBatch = typeDescription.createRowBatch(this.batchSize);
+    this.deepCleanBatch = properties.getPropAsBoolean(ORC_WRITER_DEEP_CLEAN_EVERY_BATCH, false);
+
```

Comment:
Can you log the ORC writer configuration? Useful for debugging, particularly, when topic-specific configurations are used.